### ISSUE TITLE
Clarify the regular expression use for search identifiers

### ISSUE THREAD

**Issue by 59e5aaf4** [Neo23x0/sigma#401]
The specification mentions use of regular expressions and wildcards:

> - You can use wildcard characters '*' and '?' in strings
> - Wildcards can be escaped with \, e.g. \*. If some wildcard after a backspace should be searched, the backslash has to be escaped: \\*.
> - Regular expressions are case-sensitive by default

I tried to look for use of these in the rule examples and in the source code. I understood that  `?` is only used in three windows rule examples, and the wildcard in a string (`*`) seems to be replaced by `.*` in the later conversions of the input string as a regular expression.

I saw on this comment https://github.com/Neo23x0/sigma/issues/352#issuecomment-497490331 that you lads were « working on an extension of Sigma that enables it to express regular expressions ».

The question is :
* Can we define regular expressions ?
* How does the `*` operator behave, is it specific to conversion backends ?
* How does the `?` operator behave, is it specific to conversion backends ?

The actual problem is : 
* I'd like to write IOCs matching several parameters ( let's say, `'* -NoP -sta -NonI -W Hidden -Enc *'`) whatever their order. Since YAML doesn't allow multiple keys with the same name, I can't find a way to have `CommandLine:` matching a regex resembling `(-(nop|sta|noni|w hidden|enc)){4,}` (I don't have yet the actual regex since it's not _that_ simple to build, but you get the idea: we need to have all the fields matched at least once)

A way one can avoid using regular expressions : 
* using N selections for each of the subsets, and "cluster" them with an `all of` in the `condition` rule.
```
detection:
  regexpart0: CommandLine: "*-nop*"
  regexpart1: CommandLine: "*-sta*"
  regexpart2: CommandLine: "*-noni*"
  regexpart3: CommandLine: "*-w hidden*"
  condition: all of regexpart*
```

Thanks again for the great work !

**Comment by Neo23x0** [comment]
Hi, we already did this in 
https://github.com/Neo23x0/sigma/blob/master/rules/windows/powershell/powershell_suspicious_invocation_generic.yml

So, yes, that's the best way to do it. 
The reasons is that regular expression are very CPU intensive and not all backends support it in a form that is easy or cheap to use. 

Yes, we are working on the extensions. 
I don't remember why or when we added the line `Regular expressions are case-sensitive by default` as REs cannot be used yet. 

`*` is widely used and behaves different depending on the backend. 
I haven't used `?` so far. 

Maybe @thomaspatzke can help.
